README.md

```diff
@@ -1,2 +1,597 @@
-# cli
-Clean and elegant CLI development kit.
+# {Name}
+
+[![Build](https://github.com/candy-kingdom/cli/workflows/Build/badge.svg)](https://github.com/JoshuaLight/chalice-restful/actions)
+
+_Clean and elegant CLI development kit._
+
+## Overview
+
+{Name} is a microframework that allows building declarative and nice-looking CLI apps.
+
+Consider the following example as a demonstration of {Name} capabilities:
+```py
+# say.py
+
+from {pyname} import cli, Arg, Flag, Opt
+
+@cli
+def say(phrase: Arg('a phrase to print'),
+        caps:   Flag('whether to print phrase in upper-case'),
+        times:  Opt[int]('how many times to print') = 1):
+    """Prints a phrase specified number of times."""
+
+    for i in range(0, times):
+        print(phrase.upper() if caps else phrase)
+
```

Comment:
What do you think of using a single blank space everywhere in the document?

Reply:
I think that would save a lot of space!

Reply:
Did just that.